packages/@react-spectrum/s2/src/Toast.tsx

```diff
@@ -0,0 +1,579 @@
+  let stringFormatter = useLocalizedStringFormatter(intlMessages, '@react-spectrum/s2');
+
+  // When not expanded, use a presentational div for the toasts behind the top one.
+  // The content is invisible, all we show is the background color.
+  if (!isMain && ctx && !ctx.isExpanded) {
+    return (
+      <div
+        role="presentation"
+        style={{
+          position: 'absolute',
+          [placement === 'top' ? 'bottom' : 'top']: 0,
+          left: 0,
+          width: '100%',
+          translate: `0 0 ${(-12 * index)}px`,
+          // Only 3 toasts are visible in the stack at once, but all toasts are in the DOM.
+          // This allows view transitions to smoothly animate them from where they would be 
+          // in the collapsed stack to their final position in the expanded list.
+          opacity: index >= 3 ? 0 : 1,
+          zIndex: visibleToasts.length - index - 1,
+          // When reduced motion is enabled, use append index to view-transition-name
+          // so that adding/removing a toast cross fades instead of transitioning the position.
+          // This works because the toasts are seen as separate elements instead of the same one when their index changes.
+          viewTransitionName: toast.key + (props.reduceMotion ? '-' + index : ''),
+          viewTransitionClass: [toastCss.toast, toastCss['background-toast']].map(c => CSS.escape(c)).join(' ')
+        }}
+        className={toastCss.toast + toastStyle({variant: toast.content.variant || 'info', index, isExpanded})} />
+    );
+  }
+
+  return (
+    <Toast
+      ref={toastRef}
+      toast={toast}
+      style={{
+        zIndex: visibleToasts.length - index - 1,
+        viewTransitionName: toast.key,
+        viewTransitionClass: [toastCss.toast, !isMain ? toastCss['background-toast'] : '', toastCss[placement], toastCss[align]].filter(Boolean).map(c => CSS.escape(c)).join(' ')
+      }}
+      className={renderProps => toastCss.toast + mergeStyles(
+        toastFocusRing({...renderProps, isExpanded}),
+        toastStyle({
+          variant: toast.content.variant || 'info',
+          index,
+          isExpanded
+        })
+      )}>
+      <div role="presentation" className={toastBody({isSingle: !isMain || visibleToasts.length <= 1 || isExpanded})}>
+        <ToastContent className={toastContent + (ctx && isMain ? ` ${toastCss['toast-content']}` : null)}>
+          {Icon &&
+            <CenterBaseline>
+              <Icon />
+            </CenterBaseline>
+          }
+          <Text slot="title">{toast.content.children}</Text>
+        </ToastContent>
+        {!isExpanded && visibleToasts.length > 1 && 
+          <ActionButton
+            isQuiet
+            staticColor="white"
+            styles={style({gridArea: 'expand'})}
+            // Make the chevron line up with the toast text, even though there is padding within the button.
+            UNSAFE_style={{marginInlineStart: variant === 'neutral' ? -10 : 14}}
```

Comment:
Roughly lined up? 
![Screenshot 2025-04-09 at 4 08 57 pm](https://github.com/user-attachments/assets/3d912a3f-1761-4193-aea6-87e778720766)
Looks farther off on iphone mobile for the Neutral toast

Instead of doing this as an UNSAFE_style, we could wrap the ActionButton in a div, use that for the grid area and the negative margin, it'd then get all the right scaling, and we could do away with the unsafe's?